apps/obsidian/src/components/SearchBar.tsx

```diff
@@ -0,0 +1,224 @@
+import { App, debounce } from "obsidian";
+import React, { useEffect, useState, useMemo, ReactNode } from "react";
+import { QueryEngine } from "~/services/QueryEngine";
+
+type SearchBarBaseProps<T> = {
+  app: App;
+  placeholder?: string;
+  onNodeSelect: (item: T | null) => void;
+  debounceMs?: number;
+  minQueryLength?: number;
+  getItemText: (item: T) => string;
+  getItemKey: (item: T) => string;
+  renderItemContent?: (item: T, isSelected?: boolean) => ReactNode;
+  itemStyle?: React.CSSProperties;
+  containerStyle?: React.CSSProperties;
+  inputStyle?: React.CSSProperties;
+  resultsContainerStyle?: React.CSSProperties;
```

Comment:
Remove.  Same as above.  It could be the case these are never used, for example.